aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -138,14 +138,16 @@ void set_axes_and_shapes(const Tensor& input_t,
 }
 
 void reduction_out_mps
-   (const Tensor& input_t,
+   (const Tensor& input_tensor,
     OptionalIntArrayRef opt_dim,
     bool keepdim,
     c10::optional<ScalarType> dtype,
     const Tensor& output_t,
     MPSReductionType reduction_type,
     const std::string& func_name) {
 
+  auto input_t = (input_tensor.sizes().size() == 0) ? input_tensor.view({1}) : input_tensor;
```

Comment:
If there are no elements in the tensor, can't we return early? E.g similar as you did for `norm_out_mps`:
```
 if (input_tensor.numel() == 0)
    return;
```

Reply:
@abhudev , is it expected to return `view({1})` op here?

Reply:
Sorry for the late reply! This is a different check - input_tensor.sizes().size() checks if the tensor is zero-dimensional, not if the tensor has zero elements. Zero-dimensional tensor means it is a scalar with exactly one element.

Reply:
And view({1}) treats the scalar as a tensor with shape [1], because we assume that the tensor shape has at least one element.